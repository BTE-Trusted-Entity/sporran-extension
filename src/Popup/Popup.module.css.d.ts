declare namespace PopupModuleCssNamespace {
  export interface IPopupModuleCss {
    container: string;
  }
}

declare const PopupModuleCssModule: PopupModuleCssNamespace.IPopupModuleCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: PopupModuleCssNamespace.IPopupModuleCss;
};

export = PopupModuleCssModule;
