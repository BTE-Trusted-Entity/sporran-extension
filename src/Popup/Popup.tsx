import { useEffect } from 'react';

import { browser } from 'webextension-polyfill-ts';

import { Hello } from '../components/Hello/Hello';
import { Scroller } from '../components/Scroller/Scroller';
import { CreateAccount } from '../screens/CreateAccount/CreateAccount';

import styles from './Popup.module.css';

export function Popup(): JSX.Element {
  // Sends the `popupMounted` event
  useEffect(() => {
    browser.runtime.sendMessage({ popupMounted: true });
  }, []);

  // Renders the component tree
  return (
    <div className={styles.container}>
      <div className="container mx-4 my-4">
        <Hello />
        <hr />
        <Scroller />
        <CreateAccount />
      </div>
    </div>
  );
}
