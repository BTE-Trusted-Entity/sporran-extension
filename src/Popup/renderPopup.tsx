import { render } from 'react-dom';
import { browser } from 'webextension-polyfill-ts';

import { Popup } from './Popup';

(async () => {
  await browser.tabs.query({ active: true, currentWindow: true });
  render(<Popup />, document.getElementById('popup'));
  document.documentElement.lang = browser.i18n.getUILanguage();
})();
