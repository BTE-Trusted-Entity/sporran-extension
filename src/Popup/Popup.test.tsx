import { render } from '@testing-library/react';
import { Identity, init } from '@kiltprotocol/core';

import { Popup } from './Popup';

jest.mock('@kiltprotocol/core');
(init as jest.Mock).mockImplementation(async () => 1);
(Identity.generateMnemonic as jest.Mock).mockImplementation(() => 'mnemonic');

describe('Popup', () => {
  it('should render', async () => {
    const { container, findByText } = render(<Popup />);
    await findByText(/mnemonic/);
    expect(container).toMatchSnapshot();
  });
});
