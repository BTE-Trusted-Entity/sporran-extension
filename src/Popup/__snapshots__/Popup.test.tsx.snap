// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders 1`] = `
<div
  className="container"
>
  <div
    className="container mx-4 my-4"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12 text-center"
      >
        <p
          className="text"
        >
          Example Extension «
          Sporran
          »
        </p>
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <button
          className="btn btn-block btn-outline-dark"
          onClick={[Function]}
        >
          Scroll To Top
        </button>
        <button
          className="btn btn-block btn-outline-dark"
          onClick={[Function]}
        >
          Scroll To Bottom
        </button>
      </div>
    </div>
    <p>
      
    </p>
    <p>
      
    </p>
    <form
      onSubmit={[Function]}
    >
      <label>
        Enter password:
        <input
          onChange={[Function]}
          required={true}
          type="password"
        />
      </label>
      <button
        disabled={true}
        type="submit"
      >
        Create Account
      </button>
    </form>
  </div>
</div>
`;
